### PR TITLE
Move Jetpack menu position for atomic wp-admin

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes https://github.com/Automattic/dotcom-forge/issues/5272

## Proposed changes:

| Before  | After |
| :-------------: | :-------------: |
|   | |

Self-hosted site Jetpack menu position
self-hosted

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Follow the Instructions to test on Atomic/Simple/Jetpack sites: https://github.com/Automattic/jetpack/pull/35154#issuecomment-1903704625
* Check if the menu is in the correct position for Atomic with the Classic style setting
* Check for other types of sites to see if anything breaks

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -328,6 +328,20 @@ public function get_upsell_nudge() {
...
 	}
 
+	/**
+	 * Adds Jetpack menu.
+	 *
+	 * @param int  $position The position of the menu item in the menu.
+	 * @param bool $seperator Whether to add a separator before the menu item.
+	 */
+	public function add_jetpack_menu( $position = 2, $seperator = false ) {
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
```
Comment (candy02058912) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: This seems to return false even if I set to Classic view.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 2]
```diff
@@ -328,6 +328,20 @@ public function get_upsell_nudge() {
...
+			parent::add_jetpack_menu( $position, $seperator );
+		} else {
+			parent::add_jetpack_menu();
```
Comment (candy02058912) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:339]: ~Will need to change this line to `parent::add_jetpack_menu( $position, $seperator );` to see the changes mentioned in the PR description.~

Update: I've completely changed the approach as modifying the parameters as mentioned in the strikethrough will also mean a change is needed for https://github.com/Automattic/wc-calypso-bridge/blob/master/includes/class-wc-calypso-bridge-ecommerce-admin-menu.php

**Comment by candy02058912** [comment]
~I am aware there is a typo seperator -> separator, which I'll update.~
In the meantime, I wanted to get this PR up here to get a review of whether this approach is good or not.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php (hunk 1 above)
Comment (candy02058912) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Seems like something was broken with my test site: p1706004199390829/1706004045.925339-slack-C06DN6QQVAQ

**Review by lupus2k — DISMISSED** [review]
It works great and nothing is broken outside wp-admin interface.

**Review by DustyReagan — DISMISSED** [review]
Works great!

Just throwing out an option to consider: we could create a new public method in the parent class that takes the position as a parameter to reduce duplicated code. Then we'd call that method in the base parent method and in our method override, so our override would look something like:

```
public function add_jetpack_menu() {
	// This is supposed to be the same as class-admin-menu but with a different position specified for the Jetpack menu.
	if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
		parent::create_jetpack_menu( 2 );
	} else {
		parent::add_jetpack_menu();
	}
}
```

I'm not sure if we want to do that; I'm just always looking for ways to dedupe code. 😂

**Review by taipeicoder — APPROVED** [review]
Works great 👍 

| Before | After | 
| --- | --- |
| Screenshot 2024-01-26 at 5 36 18 PM | Screenshot 2024-01-26 at 5 36 58 PM |

**Pull request merged**